internal/api/user.go

```diff
@@ -122,37 +98,84 @@ func (a *API) UserUpdate(w http.ResponseWriter, r *http.Request) error {
 
 	user := getUser(ctx)
 	session := getSession(ctx)
-	log := observability.GetLogEntry(r)
-	log.Debugf("Checking params for token %v", params)
 
-	if err := params.Validate(a.db, user, aud, config); err != nil {
+	if err := a.validateUserUpdateParams(ctx, params); err != nil {
 		return err
 	}
 
-	err = db.Transaction(func(tx *storage.Connection) error {
-		var terr error
-		if params.Password != nil {
-			if config.Security.UpdatePasswordRequireReauthentication {
-				now := time.Now()
-				// we require reauthentication if the user hasn't signed in recently in the current session
-				if session == nil || now.After(session.CreatedAt.Add(24*time.Hour)) {
-					if len(params.Nonce) == 0 {
-						return badRequestError("Password update requires reauthentication")
-					}
-					if terr = a.verifyReauthentication(params.Nonce, tx, config, user); terr != nil {
-						return terr
-					}
+	if params.AppData != nil {
```

Comment:
/nit can consider merging the 2 if checks
```suggestion
	if params.AppData != nil  && !isAdmin(user, config) {
```
since it doesn't seem like there's an alternative case to handle if user is admin